pkg/statuschecker/ready_checker.go

```diff
@@ -88,12 +88,14 @@ func New(store CRDStore, kts ...map[schema.GroupKind]ObjectStatusChecker) (*Chec
 	}, nil
 }
 
-func (c *Checker) CheckStatus(obj *unstructured.Unstructured) (ObjectStatusResult, error) {
+func (c *Checker) CheckStatus(obj *unstructured.Unstructured) ObjectStatusResult {
 	gvk := obj.GroupVersionKind()
 	gk := gvk.GroupKind()
 
 	if gk.Kind == "" || gvk.Version == "" { // Group can be empty e.g. built-in objects like ConfigMap
-		return ObjectStatusUnknown{}, errors.Errorf("object has empty kind/version: %s", gvk)
+		return ObjectStatusError{
+			Error: errors.Errorf("object has empty kind/version: %s", gvk),
```

Comment:
This is an internal error

Reply:
Hm, I'm not sure it always is - I don't remember if we validate objects in the bundle. But I'm sure I added this check to catch internal bugs and it helped quite a few times. Can you please check if we validate that objects in the Bundle have these fields set correctly.

Reply:
Yeah it's marked as internal - the ObjectStatusError also has an "IsExternalError" field